### DISCUSSION TITLE
Certificate Transparency

### DISCUSSION THREAD

**Discussion by PawlikMichal25** [cfug/dio#1718] [Ideas]
### Request Statement

I'm not an expert on this topic, but I understand that because of the problems with Certificate Pinning (eg. trouble when there is need to replace certificates, but users still have an old version of the app installed), people came up with [Certificate Transparency](https://certificate.transparency.dev/)

I know it's supported by eg. [OkHttp](https://github.com/appmattus/certificatetransparency/blob/main/docs/okhttp.md), so it would be good if Dio supported it as well. 

Related issue: https://github.com/dart-lang/sdk/issues/43352

**Comment by chawino** [comment]
I hope that this will be added too.

**Comment by Lexycon** [comment]
There is a basic implementation for this issue here:
[flutter_certificatetransparency](https://github.com/kevinasurjadi/flutter_certificatetransparency)

However, ofc this should be added in the near future in dio itself.
